examples/tree_2d_dgsem/elixir_mhd_ec_float32.jl

```diff
@@ -0,0 +1,64 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the compressible ideal GLM-MHD equations
+
+equations = IdealGlmMhdEquations2D(1.4f0)
+
+RealT = Float32
+
+initial_condition = initial_condition_weak_blast_wave
+
+volume_flux = (flux_hindenlang_gassner, flux_nonconservative_powell)
+solver = DGSEM(polydeg = 3, RealT = RealT,
+               surface_flux = (flux_hindenlang_gassner, flux_nonconservative_powell),
+               volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
+
+coordinates_min = (-2.0f0, -2.0f0)
+coordinates_max = (2.0f0, 2.0f0)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 4,
+                n_cells_max = 10_000,
+                RealT = RealT)
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0f0, 0.4f0)
```

Comment:
It's a bit confusing that you use the global `RealT` in some parts but not everywhere - and hardcode `Float32` here.

Reply:
Hm, would you suggest using `convert`s all over the place?

Reply:
It would be easier to understand if you used `RealT` everywhere - or hardcoded `Float32` values. I don't have a strong opinion on which option to choose. IIRC, I have used the latter option previously.